### PR TITLE
Update google dependencies in repository-gcs and discovery-gce

### PR DESCRIPTION
**Pull request by cwperks**
### Description

Updates various google related dependencies:

1. Update all instances of `google_http_client` to 1.44.1
2. com.google.apis:google-api-services-compute from v1-rev160-1.23.0 to v1-rev235-1.25.0
3. com.google.api-client:google-api-client from 1.23.0 to 1.35.2

### Related Issues

Discovered outdated dependencies while cleaning up the CHANGELOG for 3.0.

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwperks** [comment]
> Tests with failures:
 - org.opensearch.index.engine.NRTReplicationEngineTests.testGetSegmentInfosSnapshotPreservesFilesUntilRelease

File: plugins/discovery-gce/build.gradle [hunk 1]
```diff
@@ -17,22 +17,23 @@ opensearchplugin {
...
 }
 
-versions << [
-  'google': '1.23.0'
-]
-
 dependencies {
-  api "com.google.apis:google-api-services-compute:v1-rev160-${versions.google}"
-  api "com.google.api-client:google-api-client:${versions.google}"
+  api "com.google.apis:google-api-services-compute:v1-rev235-1.25.0"
```
Comment (reta) [inline plugins/discovery-gce/build.gradle:21]: @cwperks is `${versions.google}` used for anything now?

Comment (cwperks) [reply plugins/discovery-gce/build.gradle:21]: @reta See the lines above that were removed. `versions.google` is not coming from `buildSrc/version.properties`

Comment (reta) [reply plugins/discovery-gce/build.gradle:21]: Ah indeed, thanks you!

File: plugins/discovery-gce/build.gradle [hunk 2]
```diff
@@ -17,22 +17,23 @@ opensearchplugin {
...
+  api "com.google.http-client:google-http-client-gson:${versions.google_http_client}"
+  api "com.google.http-client:google-http-client-jackson2:${versions.google_http_client}"
   api 'com.google.code.findbugs:jsr305:3.0.2'
   api "org.apache.httpcomponents:httpclient:${versions.httpclient}"
   api "org.apache.httpcomponents:httpcore:${versions.httpcore}"
   api "commons-logging:commons-logging:${versions.commonslogging}"
   api "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
   api "commons-codec:commons-codec:${versions.commonscodec}"
+  api 'io.grpc:grpc-api:1.57.2'
+  api 'io.opencensus:opencensus-api:0.31.1'
```
Comment (reta) [inline plugins/discovery-gce/build.gradle:34]: `io.opencensus` is suspicious, this library is abandoned (part of OTEL), why do we need it here?

**Review by reta — APPROVED** [review]

**Comment by dblock** [comment]
> ❌ Gradle check result for [c7e5ee0](https://github.com/opensearch-project/OpenSearch/commit/c7e5ee0c1fe0ad3eb3a8098e97c394088871a2d6): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/37160/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

[org.opensearch.action.support.clustermanager.TransportClusterManagerNodeActionTests.testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions](https://build.ci.opensearch.org/job/gradle-check/37160/testReport/junit/org.opensearch.action.support.clustermanager/TransportClusterManagerNodeActionTests/testDontAllowSwitchingCompatibilityModeForClusterWithMultipleVersions/)

https://github.com/opensearch-project/OpenSearch/issues/13240

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@cwperks will need a manual backport please

**Comment by cwperks** [comment]
@dblock Here's the backport PR: https://github.com/opensearch-project/OpenSearch/pull/13271

**Pull request merged**